internal/corazawaf/transaction.go

```diff
@@ -900,17 +900,21 @@ func (tx *Transaction) ProcessRequestBody() (*types.Interruption, error) {
 		return nil, nil
 	}
 
-	if tx.LastPhase >= types.PhaseRequestBody {
-		// Phase already evaluated
-		tx.WAF.Logger.Warn("ProcessRequestBody has already been called")
-		return tx.interruption, nil
-	}
-
 	if tx.interruption != nil {
 		tx.WAF.Logger.Error("Calling ProcessRequestBody but there is a preexisting interruption")
 		return tx.interruption, nil
 	}
 
+	if tx.LastPhase != types.PhaseRequestHeaders {
+		if tx.LastPhase >= types.PhaseRequestBody {
+			// Phase already evaluated or skipped
+			tx.WAF.Logger.Warn("ProcessRequestBody should have already been called")
+		} else {
+			tx.WAF.Logger.Warn("Skipping request body processing, anomalous call before request headers evaluation")
```

Comment:
IIUC, this will happen regularly with Envoy. If we have a warning message here, we will still need the check on the connector side as well since we don't want to spam warning logs for correct behavior. What do you think?

Reply:
I faced it when Envoy denies on its own the request, before taking into account the Wasm filter so it was some kind of wrong behavior for Envoy. But I agree that if Envoy is already taking care of it, there is no need to generate a Warning. Moving it to Debug, so it might avoid further changes connector side, but still, show up if debugging corner cases wondering how the Wasm filter is behaving. Does it sound good?